connector/connect/client/go/go.mod

```diff
@@ -0,0 +1,53 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+module github.com/apache/spark/connector/connect/client/go/v_3_4
+
+go 1.19
```

Comment:
it seems the latest version is 1.20?

Reply:
Right, Go latest version is 1.20. It is released on Fed this year, afraid it is too new to be used by majority Go developers. Spark does not use latest version of Java, thus better not use latest Go version either?

Reply:
We generally just consider the LTS version for Java,  Spark supports three LTS Java versions(8/11/17 and 17 is the latest LTS currently) and always build and tests using the latest minor version of each LTS, so I think Spark uses the latest version of Java.

However, I'm not familiar with the version mechanism of the Go language so far, so I can't provide any better suggestions for choosing the test version at the moment. @hiboyang So can you provide more other background information? For example, which version are the mainstream Go project communities more inclined to support now?

Reply:
Got it, thanks @LuciferYang for explaining! There seems no single Go version used by mainstream Go communities. I checked several popular Go projects, they use different versions like following:

```
grpc/grpc-go go 1.17
etcd-io/etcd go 1.19
kubernetes/kubernetes go 1.20
kubernetes/client-go go 1.20
helm/helm go 1.19
```

Maybe we should make Spark Connect Go Client support and tested with multiple Go versions. Since grundprinzip is working on the CI, let's revisit this after that?